### ISSUE TITLE
1d_wave_equation bad alloc

### ISSUE THREAD

**Issue by brycelelbach** [STEllAR-GROUP/hpx#419]
**[reported by manderson]** _[Trac time Mon Jun  4 19:03:34 2012]_ (from Nathan Addy (naddy@lbl.gov)

I'm having trouble running all the examples in examples.quickstart,
notably the 1d_wave_equation example.

Basically a bad_alloc gets thrown and isn't caught.  It appears to
occur in the line which waits for all the futures to return:

wait(futures, [&](std::size_t i, double n)
        { double x_here = i*dx;
        outfile << (boost::format("%1% %2%\n") % x_here % n) << flush; });

on approximately line 240.

It doesn't have anything to do with the lambda: the problem persists
even with a null body.

I don't really have an understanding of the problem and plan to check
it out further.  By default, my best guess is that it's a 64 bit
issue.  However, the other quickstart examples work and all tests of
libunwind/gperftools pass.  Additionally, the problem seems to persist
when compiling using the system malloc.

I am using gcc4.6.1, Boost v1.47.0, hpx 0.8.1.  The error is:
[what]: std::bad_alloc: HPX(unhandled_exception)
[version]: V0.8.1 (AGAS: V2.1), SVN: exported

[date]: Jun  4 2012 11:07:51

[compiler]: GNU C++ version 4.6.1
[stdlib]: GNU libstdc++ version 20110627

I am currently trying to track this down, by rebuilding on a 32 bit
machine, but am not having much luck.  Is this a problem you've
encountered before?  Any suggestions as to what I might try or
anything else that might give me a push in the right direction?

I suspect this is really critical for us.  As near as I can tell, all
the quickstart examples run except this one.  However, the
big/non-quickstart examples (eg graph500, heartbeat,
managed_accumulator_client, etc) all fail with either bad allocations
or with segfaults or other things like that.

Basically, any insight would be greatly appreciated.
